### PR TITLE
Trojita: PGP decryption support

### PR DESCRIPTION
**Pull request by ghost**
###### Motivation for this change

Every email client should have some sort of support for PGP. Trojitá is currently decrypt-only.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @FRidh

### CODE REVIEW

**Comment by peterhoeg** [comment]
I didn't see this - sorry. Does #68540 solve it for you?

File: pkgs/applications/networking/mailreaders/trojita/default.nix [hunk 1]
```diff
@@ -16,21 +19,26 @@ mkDerivation rec {
     sha256 = "1n9n07md23ny6asyw0xpih37vlwzp7vawbkprl7a1bqwfa0si3g0";
   };
 
+  patches = [ ./keygen.sh.patch ];
```
Comment (Mic92) [inline pkgs/applications/networking/mailreaders/trojita/default.nix]: Can you send this patch upstream as well?

File: pkgs/development/libraries/kf5gpgmepp/default.nix [hunk 2]
```diff
@@ -0,0 +1,24 @@
+{ stdenv, lib, fetchgit, cmake, kdeFrameworks, qt5, boost, gpgme }:
+
+stdenv.mkDerivation rec {
+  pname = "kf5gpgmepp";
+  version = "16.08.3";
+
+  src = fetchgit {
+    url = "git://anongit.kde.org/gpgmepp.git";
```
Comment (Mic92) [inline pkgs/development/libraries/kf5gpgmepp/default.nix]: Https is preferred here: https://anongit.kde.org/gpgmepp.git

**Comment by ehmry** [comment]
> I didn't see this - sorry. Does #68540 solve it for you?

No.

File: pkgs/development/libraries/kf5gpgmepp/default.nix [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+  buildInputs = [ kdeFrameworks.extra-cmake-modules qt5.qtbase boost ];
+  propagatedBuildInputs = [ gpgme ];
+
+  nativeBuildInputs = [ cmake ];
+
+  meta = {
+    license = with lib.licenses; [ lgpl2 ];
+    maintainers = with lib.maintainers; [ ehmry ];
+    platforms = with lib.platforms; linux;
```
Comment (Mic92) [inline pkgs/development/libraries/kf5gpgmepp/default.nix]: Does it has a homepage?

**Comment by ehmry** [comment]
Rebased on #68540, the keygen.sh patch is gone.

File: pkgs/top-level/all-packages.nix [hunk 4]
```diff
@@ -11494,6 +11494,8 @@ in
     automake = automake111x;
   };
 
+  kf5gpgmepp = callPackage ../development/libraries/kf5gpgmepp { };
```
Comment (peterhoeg) [inline pkgs/top-level/all-packages.nix]: this should be ```libsForQt5.callPackage```

File: pkgs/development/libraries/kf5gpgmepp/default.nix [hunk 5]
```diff
@@ -0,0 +1,24 @@
+{ stdenv, lib, fetchgit, cmake, kdeFrameworks, qt5, boost, gpgme }:
+
+stdenv.mkDerivation rec {
```
Comment (peterhoeg) [inline pkgs/development/libraries/kf5gpgmepp/default.nix]: With the libsForQt5 stuff below, this should become mkDerivation.

File: pkgs/development/libraries/kf5gpgmepp/default.nix [hunk 6]
```diff
@@ -0,0 +1,24 @@
+{ stdenv, lib, fetchgit, cmake, kdeFrameworks, qt5, boost, gpgme }:
```
Comment (peterhoeg) [inline pkgs/development/libraries/kf5gpgmepp/default.nix]: qt5 should then go as well (and kdeFrameworks). Then add qtbase and extra-cmake-modules.

**Review by peterhoeg — CHANGES_REQUESTED** [review]

File: pkgs/development/libraries/kf5gpgmepp/default.nix (hunk 3 above)
Comment (ehmry) [reply pkgs/development/libraries/kf5gpgmepp/default.nix]: No homepage that I could find.

File: pkgs/development/libraries/kf5gpgmepp/default.nix [hunk 7]
```diff
@@ -0,0 +1,24 @@
+{ stdenv, lib, fetchgit, cmake, extra-cmake-modules, qtbase, boost, gpgme }:
+
+stdenv.mkDerivation {
+  pname = "kf5gpgmepp";
+  version = "16.08.3";
+
+  src = fetchgit {
+    url = "https://anongit.kde.org/gpgmepp.git";
```
Comment (Lassulus) [inline pkgs/development/libraries/kf5gpgmepp/default.nix]:
```suggestion
    url = "https://cgit.kde.org/gpgmepp.git";
```

File: pkgs/applications/networking/mailreaders/trojita/default.nix [hunk 8]
```diff
@@ -1,8 +1,10 @@
 { akonadi-contacts
 , cmake
 , fetchgit
+, gnupg
 , gpgme
 , kcontacts
+, kf5gpgmepp
 , lib
```
Comment (Lassulus) [inline pkgs/applications/networking/mailreaders/trojita/default.nix]: remove lib here and use `with stdenv.lib` further down

File: pkgs/development/libraries/kf5gpgmepp/default.nix [hunk 9]
```diff
@@ -0,0 +1,24 @@
+{ stdenv, lib, fetchgit, cmake, extra-cmake-modules, qtbase, boost, gpgme }:
```
Comment (Lassulus) [inline pkgs/development/libraries/kf5gpgmepp/default.nix]:
```suggestion
{ stdenv, fetchgit, cmake, extra-cmake-modules, qtbase, boost, gpgme }:
```

File: pkgs/development/libraries/kf5gpgmepp/default.nix [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+    rev = "9826f6674e496ce575f606d17c318566381b3b15";
+    sha256 = "02ck2l3s8s7xh44blqaqnc5k49ccicdnzvhiwa67a3zgicz5i0vh";
+  };
+
+  buildInputs = [ extra-cmake-modules qtbase boost ];
+  propagatedBuildInputs = [ gpgme ];
+
+  nativeBuildInputs = [ cmake ];
+
+  meta = {
```
Comment (Lassulus) [inline pkgs/development/libraries/kf5gpgmepp/default.nix]:
```suggestion
  meta with stdenv.lib; = {
```

**Pull request merged**